CMakeLists.txt

```diff
@@ -17,60 +22,252 @@ if(MSVC)
   string(REGEX REPLACE " /M[TD]d?" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 endif()
 
+if(NOT GENERATOR_IS_MULTI_CONFIG AND CMAKE_BUILD_TYPE STREQUAL "")
+  message(STATUS "Defaulting to Debug build as no build type was specified")
+  set(CMAKE_BUILD_TYPE "Debug")
+endif()
+
 if(WIN32)
   # this is needed, because Windows can be weird, man.
   set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreadedDLL")
 endif()
 
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
+# Prepend so that our FindVulkan gets picked up first
+list(PREPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 
+option(NOVELRT_FETCH_DEPENDENCIES "Fetch and build NovelRT dependencies instead of providing system-based ones" ON)
 option(NOVELRT_BUILD_SAMPLES "Build NovelRT samples" ON)
-option(NOVELRT_BUILD_DOCUMENTATION "Build NovelRT documentation" ON)
+option(NOVELRT_BUILD_DOCUMENTATION "Build NovelRT documentation" OFF)
 option(NOVELRT_BUILD_TESTS "Build NovelRT tests" ON)
 option(NOVELRT_BUILD_INTEROP "Build NovelRT's Interop library" ON)
+option(NOVELRT_VERBOSE_BUILD "Build NovelRT using verbose output" OFF)
+option(NOVELRT_INSTALL "Generate installation targets" ON)
 
-set(NOVELRT_BZIP2_VERSION "1.0.8" CACHE STRING "BZip2 version")
 set(NOVELRT_DOXYGEN_VERSION "1.8.17" CACHE STRING "Doxygen version")
-set(NOVELRT_FREETYPE_VERSION "2.12.1" CACHE STRING "Freetype version")
-set(NOVELRT_GSL_VERSION "4.0.0" CACHE STRING "ms-gsl version")
+set(NOVELRT_FLAC_VERSION "1.3.4" CACHE STRING "FLAC version")
+set(NOVELRT_GLFW_VERSION "3.3.7" CACHE STRING "GLFW3 version")
+set(NOVELRT_GSL_VERSION "4.0.0" CACHE STRING "Microsoft.GSL version")
 set(NOVELRT_ONETBB_VERSION "2021.5.0" CACHE STRING "OneTBB version")
 set(NOVELRT_OPENAL_VERSION "1.21.1" CACHE STRING "OpenAL version")
+set(NOVELRT_OGG_VERSION "1.3.5" CACHE STRING "Ogg version")
+set(NOVELRT_OPUS_VERSION "1.3.1" CACHE STRING "Opus version")
 set(NOVELRT_PNG_VERSION "1.6.35" CACHE STRING "PNG version")
-set(NOVELRT_VULKAN_VERSION "1.3.211" CACHE STRING "Vulkan version")
```

Comment:
What happened here?

Reply:
It just got moved to line 58, and version bumped to 1.3.231.1